### PR TITLE
Add Sys.is<os>

### LINKED ISSUE #378: Support 0.7-style Sys.is<os> rather than is_<os>
I'm not sure how best to do this, since it involves a Base module that as far as I know has been in every version, unlike Iterators. For reference, this was introduced in 0.7.0-DEV.914.

### PR DESCRIPTION
**Pull request by ararslan**
Fixes #378.

### CODE REVIEW

**Comment by ararslan** [comment]
Sorry @yuyichao, I wasn't sufficiently diligent in my initial iteration here. You may want to have another look to ensure this still meets expectations.

**Comment by ararslan** [comment]
Gah, never mind, it's still wrong for 0.5 and 0.6...

**Comment by ararslan** [comment]
Alright, I think things are good now. The failure on nightly is unrelated.

File: test/runtests.jl [hunk 1]
```diff
@@ -1242,6 +1242,14 @@ if VERSION < v"0.5.0-dev+4267"
...
     @test Compat.KERNEL == Sys.KERNEL
+    for os in [:apple, :bsd, :linux, :unix, :windows]
+        from_base = if VERSION >= v"0.7.0-DEV.914"
+            Expr(:., Expr(:., :Base, Base.Meta.quot(:Sys)), Base.Meta.quot(Symbol("is", os)))
+        else # VERSION >= v"0.5.0-dev+4267"
+            Expr(:., :Base, Base.Meta.quot(Symbol("is_", os)))
+        end
+        @eval @test Compat.Sys.$(Symbol("is", os))() == $from_base()
+    end
 end
```
Comment (Sacha0) [inline test/runtests.jl]: Do I understand correctly that the objective is to support `Sys.is$(os)` from 0.5 on only? If so I will lgtm :).

Comment (ararslan) [inline test/runtests.jl]: Yes, though 0.4 support will soon be dropped here anyway.

**Comment by tkelman** [comment]
Should have actually read the logs, this is giving `WARNING: imported binding for Sys overwritten in module Compat` on 0.5

**Comment by tkelman** [comment]
0.6 as well

**Comment by tkelman** [comment]
This probably should eval into `Base.Sys` to avoid that warning

**Comment by ararslan** [comment]
I had initially suggested evaling in #378. I can change it to that.

**Pull request merged**